client/ayon_core/hosts/substancepainter/api/lib.py

```diff
@@ -640,3 +642,58 @@ def _setup_prompt():
         return
 
     return project_mesh
+
+
+def get_export_presets_by_filtering(export_preset_name, channel_type_list):
+    new_maps = []
+
+    export_presets = get_export_presets()
+    resource_presets = substance_painter.export.list_resource_export_presets()
+    preset = next((preset for preset in resource_presets
+                   if preset.resource_id.name == (
+                       export_presets[export_preset_name])), None)
```

Comment:
I wonder if this is more readable.
```suggestion
    export_presets = get_export_presets()
    export_preset_nice_name = export_presets[export_preset_name]
    resource_presets = substance_painter.export.list_resource_export_presets()
    preset = next(
        (
            preset for preset in resource_presets
            if preset.resource_id.name == export_preset_nice_name
        ), None
    )
```